lib/elixir/lib/kernel.ex

```diff
@@ -795,11 +798,13 @@ defmodule Kernel do
 
       iex> trunc(5.4)
       5
+
       iex> trunc(5.99)
       5
 
   """
-  @spec trunc(number) :: integer
+  @spec trunc(arg) :: arg when arg: integer
```

Comment:
It does use when: https://github.com/erlang/otp/blob/maint/lib/stdlib/src/lists.erl#L1231-L1236